Fixes #3564  Merge profile progress header xml

bkaur-bkj opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes #3564  
merged portrait and landscape files of tab layout, could not merge the mobile layout files as they had some constraints and other differences too.
## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing
mobile landscape before 
![WhatsApp Image 2021-10-07 at 1 15 42 PM (1)](https://user-images.githubusercontent.com/64087572/136342560-77ecb9b3-4d1e-4a92-a899-368f1e930d57.jpeg)

mobile landscape after changes
![WhatsApp Image 2021-10-07 at 1 15 42 PM](https://user-images.githubusercontent.com/64087572/136342577-59c1f9f1-9eb9-491f-b98a-4430bb083c93.jpeg)


![WhatsApp Image 2021-10-07 at 1 25 32 PM (1)](https://user-images.githubusercontent.com/64087572/136343327-0272a91b-058d-4cef-a37a-de974b3c20dd.jpeg)


![WhatsApp Image 2021-10-07 at 1 25 32 PM](https://user-images.githubusercontent.com/64087572/136343339-a2a04e5c-b8fb-42b0-9f30-005ac3569a23.jpeg)


After changes tablet portrait

![Screenshot_1633716487](https://user-images.githubusercontent.com/64087572/136603193-1035e9fd-2f41-4044-8728-90a0812fbe21.png)

After changes tablet landscape

![Screenshot_1633716474](https://user-images.githubusercontent.com/64087572/136604924-9ec5baef-0255-4080-879b-d157b2c57490.png)

bkaur-bkj committed:
merged landscape and portrait tab layout xml

bkaur-bkj committed:
removed extra blank lines

bkaur-bkj commented:
@rt4914  PTAL, as discussed I made a PR merging the 2 tab layout files but it says some conflicts I tried viewing them but got no idea how to resolve them and not understood the issue.

bkaur-bkj committed:
Merge branch 'develop' into merge_profile_progress_header_xml

bkaur-bkj commented:
@rt4914, I resolved the conflicts but one test is failing

rt4914 commented:
@bkaur-bkj As mentioned in the issue description please  add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

rt4914 commented:
> @rt4914, I resolved the conflicts but one test is failing

@bkaur-bkj The test which is failing is an optional one and also it mostly is failing on develop too so don't worry about it in this PR.

bkaur-bkj committed:
changed location of profile_progress_header.xml from layout-sw600dp-port to layout-sw600dp

bkaur-bkj commented:
@rt4914 I have added ss for mobile portrait and landscape but I tried for tablet it crashes, as I tried merging the landscape and portrait layout of tablet I removed tablet landscape layout and tried running at app on tab, it was fine with portrait but crashed in landscape so I changed the folder of the file and placed the file profile_progress_header in the layout/swa600dp/ folder. But it not crases for portrait layout too.
pls let me know what change should I do

rt4914 commented:
> @rt4914 I have added ss for mobile portrait and landscape but I tried for tablet it crashes, as I tried merging the landscape and portrait layout of tablet I removed tablet landscape layout and tried running at app on tab, it was fine with portrait but crashed in landscape so I changed the folder of the file and placed the file profile_progress_header in the layout/swa600dp/ folder. But it not crases for portrait layout too. pls let me know what change should I do

@bkaur-bkj For tablet follow these steps:
1. Uninstall the app
2. `Build` -> `Rebuild` the project.
3. Now install the app.
Basically whenever we remove any xml file for tablet we need to uninstall the previous version and install the fresh build.

rt4914 commented:
@bkaur-bkj Also please merge with latest develop as there are conflicts currently.

bkaur-bkj committed:
Merge branch 'develop' into merge_profile_progress_header_xml

bkaur-bkj commented:
> @bkaur-bkj Also please merge with latest develop as there are conflicts currently.

Yes sir Done

bkaur-bkj commented:
> > @rt4914 I have added ss for mobile portrait and landscape but I tried for tablet it crashes, as I tried merging the landscape and portrait layout of tablet I removed tablet landscape layout and tried running at app on tab, it was fine with portrait but crashed in landscape so I changed the folder of the file and placed the file profile_progress_header in the layout/swa600dp/ folder. But it not crases for portrait layout too. pls let me know what change should I do
> 
> @bkaur-bkj For tablet follow these steps:
> 
> 1. Uninstall the app
> 2. `Build` -> `Rebuild` the project.
> 3. Now install the app.
>    Basically whenever we remove any xml file for tablet we need to uninstall the previous version and install the fresh build.

yes done worked now

rt4914 commented on app/src/main/res/values-sw600dp-land/dimens.xml:
```diff
@@ -348,6 +348,13 @@
   <dimen name="topic_lessons_fragment_layout_padding_bottom">100dp</dimen>
 
+
+  <!-- Profile Progress Header-->
```
Space before and after the text
`<!-- Profile Progress Header -->`

bkaur-bkj replied:
done

rt4914 commented on app/src/main/res/values-sw600dp-port/dimens.xml:
```diff
@@ -350,6 +350,12 @@
   <!-- Topic Lessons Fragment -->
   <dimen name="topic_lessons_fragment_layout_padding_bottom">100dp</dimen>
 
+  <!-- Profile Progress Header-->
```
Ditto

bkaur-bkj replied:
done

rt4914 commented on app/src/main/res/values/dimens.xml:
```diff
@@ -521,6 +521,12 @@
   <dimen name="story_header_view_layout_margin_start">32dp</dimen>
   <dimen name="story_header_view_layout_margin_end">32dp</dimen>
 
+  <!-- Profile Progress Header-->
```
Ditto

bkaur-bkj replied:
done

rt4914 commented on app/src/main/res/values/dimens.xml:
```diff
@@ -521,6 +521,12 @@
+  <dimen name="profile_edit_image_layout_margin_start">42dp</dimen>
+  <dimen name="profile_name_textview_layout_margin_end">202dp</dimen>
+  <dimen name="recently_played_stories_textView_layout_margin_start">42dp</dimen>
+  <dimen name="view_all_textview_margin_end">24dp</dimen>
```
These values should be `0dp`  and also add a comment on top mentioning this:
`These values are getting used only in tablet mode and therefore mentioned as 0dp here.`

bkaur-bkj replied:
done

rt4914 reviewed (changes requested):
@bkaur-bkj PTAL

bkaur-bkj committed:
added spaces before and after text and added comment for dimens.xml file

bkaur-bkj committed:
Merge branch 'develop' into merge_profile_progress_header_xml

rt4914 commented on app/src/main/res/values/dimens.xml:
```diff
@@ -521,6 +521,13 @@
+  <!-- Profile Progress Header -->
+  <dimen name="profile_edit_image_layout_margin_start">0dp</dimen>
+  <dimen name="profile_name_textview_layout_margin_end">0dp</dimen>
+  <dimen name="recently_played_stories_textView_layout_margin_start">0dp</dimen>
```
There is inconsistency in writing the word `TextView`. You have written it as `textView` as well as `textview`. 

Please use following `text_view`.

This change will affect almost all files.

bkaur-bkj replied:
okay sir

rt4914 reviewed (changes requested):
One comment left.
@bkaur-bkj PTAL

bkaur-bkj committed:
refractor textView to text_view

bkaur-bkj committed:
Resolving failed build

bkaur-bkj commented:
@rt4914 PTAL

rt4914 reviewed (approved):
LGTM, thanks.
